### PR TITLE
feat(securitycenter): Add Resource SCC Management API Org SHA Custom Module samples (Create, Get, List, Update)

### PR DESCRIPTION
**Pull request by vijaykanthm**
## Description

Fixes #b/347347419, b/347346767, b/347347451, b/347347162
Reference to b/347347419, b/347346767, b/347347451, b/347347162
This PR adds SCC Managament API Org SHA Custom Module Samples to Create, List, Get & Update.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: securitycenter/management_api/get_security_health_analytics_custom_module.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import (
+	"context"
+	"fmt"
+	"io"
+
+	securitycentermanagement "cloud.google.com/go/securitycentermanagement/apiv1"
+	securitycentermanagementpb "cloud.google.com/go/securitycentermanagement/apiv1/securitycentermanagementpb"
+)
+
+// GetSecurityHealthAnalyticsCustomModule retrieves a specific custom module by its name.
```
Comment (telpirion) [inline securitycenter/management_api/get_security_health_analytics_custom_module.go:28]: nit: same as above. Probably best to ensure that the name of the function in the comment matches the name of the function.

File: securitycenter/management_api/list_security_health_analytics_custom_module.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	"fmt"
+	"io"
+
+	securitycentermanagement "cloud.google.com/go/securitycentermanagement/apiv1"
+	securitycentermanagementpb "cloud.google.com/go/securitycentermanagement/apiv1/securitycentermanagementpb"
+	iterator "google.golang.org/api/iterator"
+)
+
+// ListSecurityHealthAnalyticsCustomModule creates a custom module for Security Health Analytics.
+func listSecurityHealthAnalyticsCustomModule(w io.Writer, parent string) error {
```
Comment (telpirion) [inline securitycenter/management_api/list_security_health_analytics_custom_module.go:30]: Same as above, about function name in comment.

File: securitycenter/management_api/update_security_health_analytics_custom_module.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	securitycentermanagement "cloud.google.com/go/securitycentermanagement/apiv1"
+	securitycentermanagementpb "cloud.google.com/go/securitycentermanagement/apiv1/securitycentermanagementpb"
+	fieldmaskpb "google.golang.org/protobuf/types/known/fieldmaskpb"
+)
+
+// UpdateSecurityHealthAnalyticsCustomModule creates a custom module for Security Health Analytics.
```
Comment (telpirion) [inline securitycenter/management_api/update_security_health_analytics_custom_module.go:29]: Same as previous.

**Review by telpirion — APPROVED** [review]
No blocking issues. Please accept/reject suggestions as you see fit.

File: securitycenter/management_api/create_security_health_analytics_custom_module.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	securitycentermanagement "cloud.google.com/go/securitycentermanagement/apiv1"
+	securitycentermanagementpb "cloud.google.com/go/securitycentermanagement/apiv1/securitycentermanagementpb"
+	expr "google.golang.org/genproto/googleapis/type/expr"
+)
+
+// CreateSecurityHealthAnalyticsCustomModule creates a custom module for Security Health Analytics.
```
[the remark this answers is not in the document]
Comment (vijaykanthm) [reply securitycenter/management_api/create_security_health_analytics_custom_module.go]: Addressed.

File: securitycenter/management_api/get_security_health_analytics_custom_module.go (hunk 1 above)
Comment (vijaykanthm) [reply securitycenter/management_api/get_security_health_analytics_custom_module.go]: Addressed.

File: securitycenter/management_api/list_security_health_analytics_custom_module.go (hunk 2 above)
Comment (vijaykanthm) [reply securitycenter/management_api/list_security_health_analytics_custom_module.go:30]: Addressed.

File: securitycenter/management_api/update_security_health_analytics_custom_module.go (hunk 3 above)
Comment (vijaykanthm) [reply securitycenter/management_api/update_security_health_analytics_custom_module.go]: Addressed.

**Pull request merged**